everest-testing/src/everest/testing/core_utils/fixtures.py

```diff
@@ -1,26 +1,104 @@
+        )
 
+    return None
+
+
+@pytest.fixture
+def core_config(request) -> EverestEnvironmentCoreConfiguration:
     everest_prefix = Path(request.config.getoption("--everest-prefix"))
+
     marker = request.node.get_closest_marker("everest_core_config")
     if marker is None:
-        everest_core = EverestCore(everest_prefix)
+        everest_config_path = None  # config auto-detected by everest core
     else:
         path = Path('/etc/everest') if everest_prefix == '/usr' else everest_prefix / 'etc/everest'
-        config_path = path / marker.args[0]
-        everest_core = EverestCore(everest_prefix, config_path)
-    yield everest_core
+        everest_config_path = path / marker.args[0]
+
+    return EverestEnvironmentCoreConfiguration(
+        everest_core_path=everest_prefix,
+        template_everest_config_path=everest_config_path,
+    )
+
+
+@pytest.fixture
+def ocpp_config(request) -> Optional[EverestEnvironmentOCPPConfiguration]:
+    return None
+
+@pytest.fixture
+def evse_security_config(request) -> Optional[EverestEnvironmentEvseSecurityConfiguration]:
+    source_certs_dir_marker = request.node.get_closest_marker("source_certs_dir")
+    if source_certs_dir_marker:
+        return EverestEnvironmentEvseSecurityConfiguration(source_certificate_directory=Path(source_certs_dir_marker.args[0]))
+    return None
+
+@pytest.fixture
+def persistent_store_config(request) -> Optional[EverestEnvironmentPersistentStoreConfiguration]:
+    persistent_store_marker = request.node.get_closest_marker("use_temporary_persistent_store")
+    if persistent_store_marker:
+        return EverestEnvironmentPersistentStoreConfiguration(use_temporary_folder=True)
+    return None
+
+
+
+@pytest.fixture
+def everest_core(request,
+                 tmp_path,
+                 core_config: EverestEnvironmentCoreConfiguration,
+                 ocpp_config: Optional[EverestEnvironmentOCPPConfiguration],
+                 probe_module_config: Optional[EverestEnvironmentProbeModuleConfiguration],
+                 evse_security_config: Optional[EverestEnvironmentEvseSecurityConfiguration],
+                 persistent_store_config: Optional[EverestEnvironmentPersistentStoreConfiguration],
+                 ) -> EverestCore:
+    """Fixture that can be used to start and stop everest-core"""
+
+    standalone_module_marker = request.node.get_closest_marker('standalone_module')
+
+    environment_setup = EverestTestEnvironmentSetup(
+        core_config=core_config,
+        ocpp_config=ocpp_config,
+        probe_config=probe_module_config,
+        evse_security_config=evse_security_config,
+        persistent_store_config=persistent_store_config,
+        standalone_module=list(standalone_module_marker.args) if standalone_module_marker else None
+    )
+
+    environment_setup.setup_environment(tmp_path=tmp_path)
+    yield environment_setup.everest_core
+
+    # FIXME (aw): proper life time management, shouldn't the fixure start and stop?
```

Comment:
Are these FIXMEs relevant?

Reply:
yes indeed... should be discussed with Kai. I would agree with his comment and actually remove the stops here.

But I don't know where this is used and this could potentially break tests.

Reply:
so to keep it a "refactoring",  I would leave this